### ISSUE TITLE
LFX Workspace: Unified WasmEdge tools

### ISSUE THREAD

**Issue by michael1017** [WasmEdge/WasmEdge#2345]
# Motivation
WasmEdge provides two tools in the release assets: `wasmedgec` and `wasmedge`. However, several tools are planned to be released, including `wasm-signature-related tools (wasmedge-sign and wasmedge-verify, a draft idea, not finalized)`, `wasmedge-test (which allows users to execute wast, wasm test script files)`, and more.

However, providing multiple tools will make it too complicated to use. That's why we want a simple entry point, `wasmedge`. As its subcommands, all the tools above should be collected into this new tool.

```
Before:
wasmedge
wasmedgec
wasmedge-sign
wasmedge-verify
wasmedge-test
After:
wasmedge, wasmedge run
wasmedge compile
wasmedge sign
wasmedge verify
wasmedge test
```

# Milestones

- [x] Check `class Options` in [wasm source file](https://github.com/WasmEdge/WasmEdge/blob/master/include/po/option.h) and learn how to use it to add new options and subcommands. 
- [x] Make changes in `class Options` and `argument_parser` if needed. (Since by default, the new options would have `--` prefix. However, we don't want to add `--` prefix like this: `wasmedge --compile`. We want `wasmedge compile`)
- [x] Start working on adding new features `wasmedge compile` and `wasmedge, wasmedge run`. The expected result is that the arguments after `wasmedge compile` would be forwarded to [Compiler](https://github.com/WasmEdge/WasmEdge/blob/master/lib/driver/compilerTool.cpp), and the arguments after `wasmedge, wasmedge run` would be forwarded to [Tool](https://github.com/WasmEdge/WasmEdge/blob/master/lib/driver/runtimeTool.cpp)
- [ ] ~~Clear up old command `wasmedgec`.~~
- [x] Update documents to demonstrate how to use the new features.

**Comment by michael1017** [comment]
# Progress Report
## Week1 Progress

1. Move options in the original tools (`wasmedge`, `wasmedgec`) into a struct. [Details](https://github.com/WasmEdge/WasmEdge/commit/5787459812efb09625c93675f4f14f2ede75bfc6) here.
2. Construct Parsers for different CAPI such as `WasmEdge_Driver_Compiler` and `WasmEdge_Driver_Tool` in file lib/driver/uniTool.cpp. [Details](https://github.com/WasmEdge/WasmEdge/commit/16f050712ff254da1eaa3058041aeaea172a779e#diff-35b899c2972542a4ab9cdfbe655426864c99d67cd064d69acf9bbe61ae0f6493) here.
3. Add new Wasmedge CAPI `WasmEdge_Driver_UniTool`  for unified tool. [Details](https://github.com/WasmEdge/WasmEdge/commit/16f050712ff254da1eaa3058041aeaea172a779e#diff-6e5f0525622a9f5e0d2855c4486743f31e5d586c26932242e58a4ccd6e9e5649) here.
4. Fix subcommand usage hint. [Details](https://github.com/WasmEdge/WasmEdge/commit/f9e9a472e7acbe6cf96680e59a7bb2de187d50e4#diff-8476816f0372790d392f3f90c65df1bfa5ab71d6ddc3caf3f74ce945fd9221b6) here.
5. Essential document update. [Details](https://github.com/WasmEdge/WasmEdge/commit/4e67b254a4ac17c91324eceea1645d8991b512f2) here.

## Week2 Plan
In the milestones I listed, the next step we should do is cleaning up old command `wasmedgec`. However, `wasmedgec` appears in several places and need to be carefully removed. Therefore, I split the mission into three parts.

1. Remove `wasmedgec` in GitHub action .yml files.
2. Remove `wasmedgec` description in CMakeLists.txt.
3. Remove `wasmedgec` in documents.

I will follow the order I listed above to remove `wasmedgec`.

**Comment by hydai** [comment]
For compatibility, we should keep the wasmedgec for several versions. It would be better to have both, the new unified tool and the existing one. But we can have a deprecated warning when users try to execute wasmedgec.

**Comment by michael1017** [comment]
> For compatibility, we should keep the wasmedgec for several versions. It would be better to have both, the new unified tool and the existing one. But we can have a deprecated warning when users try to execute wasmedgec.

Got it! I will keep the wasmedgec and add a deprecated warning fot it. Lets update our week2 plan.

1. Add deprecated warning for `wasmedgec`.
2. Add `wasmedge compile` into GitHub action .yml files. Since in some .yml files such as [build_for_riscv.yml](https://github.com/WasmEdge/WasmEdge/blob/master/.github/workflows/build_for_riscv.yml) runs `wasmedgec` for CI. I thinks it is better to add `wasmedge compile` as well.
3. `wasmedgec` doesn't need to be changed in CMakeLists.txt.
4. Now I am wondering if you wish to replace `wasmedgec` into `wasmedge compile` in the documents? If you have any ideas, please let me know.

**Comment by michael1017** [comment]
# Progress Report
## Week2 Progress
1. Update documents for wasmedge unified tool.
2. Since we reserve wasmedgec tool at this time, I did not remove wasmedgec in the installation documents. Besides, I added notes to remind user that wasmedgec is equal to wasmedge compile and we decide to deprecate wasmedgec in the future.
3. Since wasmedgec is equal to wasmedge compile, I replace wasmedgec to wasmedge compile in user guides and examples.
4. Submit draft #2378. (Once #2363 is merged into master branch, I will set #2378 as ready to review)

## Week3 Plan
1. Add deprecated warning for `wasmedgec`.
2. Add `wasmedge compile` into GitHub action .yml files. Since in some .yml files such as [build_for_riscv.yml](https://github.com/WasmEdge/WasmEdge/blob/master/.github/workflows/build_for_riscv.yml) runs wasmedgec for CI. I thinks it is better to add wasmedge compile as well.

**Comment by michael1017** [comment]
@hydai Thanks for your mentoring!!

Merged: 
#2363 
#2378 
#2385
